schema/workflow.json

```diff
@@ -1,33 +1,26 @@
 {
-  "$id": "https://serverlessworkflow.io/schemas/0.8/workflow.json",
+  "$id": "https://serverlessworkflow.io/schemas/0.9/workflow.json",
   "$schema": "http://json-schema.org/draft-07/schema#",
   "description": "Serverless Workflow specification - workflow schema",
   "type": "object",
   "properties": {
-    "id": {
-      "type": "string",
-      "description": "Workflow unique identifier",
-      "minLength": 1
-    },
-    "key": {
+    "name": {
       "type": "string",
-      "description": "Domain-specific workflow identifier",
+      "description": "Workflow definition unique identifier (which must be human readable)",
```

Comment:
Shouldn't the concatenation of both `name` and `version` be unique, instead of just the `name`?

If so, what about something like:

```
The name that identifies the workflow definition, and which, when combined with its version, forms a unique identifier. 
```

Reply:
I think you are overcomplicating it ;)
The name should be unique regardless whatever is in the version, so combined with the version it will allways be unique.

Reply:
The problem I have is with the `unique` adjective you are using. 

Given the terminology used, I definitly see how one could then confuse the `name` with the concept of an `id`: only one 'my-cool-workflow' can ever exist.

In my opinion, we should be as explicit as possible, especially in the `specification.md` file, so that authors and implementers understand that, to reference a workflow definition in a UNIQUE fashion, you should instead use `{name}:{version}`, for example. In other words, you can have multiple 'my-cool-workflow' definitions, but they must all have different versions.

Reply:
Ok, the only problem I find with your re phrasing is that version is optional, and somehow someone can understood that is not.

Reply:
Yeah, sorry about that, that was something we discussed yesterday in the weekly.

What was proposed is to provide `version` a default `latest` value, so that even if it's optional and not set by the user, a value is always supplied, making `version` non-nullable.